template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/WebApi-CSharp/Areas/IdentityService/IdentityServiceJwtBearerOptionsSetup.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity.Service;
+using Microsoft.AspNetCore.Identity.Service.IntegratedWebClient;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Company.WebApplication1.Identity
+{
+    public class JwtBearerOptionsSetup : IConfigureOptions<JwtBearerOptions>
+    {
+        public JwtBearerOptionsSetup(
+            IConfiguration config,
+            IHttpContextAccessor httpContextAccessor,
+            IKeySetMetadataProvider keysProvider)
+        {
+            Configuration = config;
+            HttpContextAccessor = httpContextAccessor;
+            KeySetMetadataProvider = keysProvider;
+        }
+
+        public IConfiguration Configuration { get; set; }
+        public IHttpContextAccessor HttpContextAccessor { get; set; }
+        public IKeySetMetadataProvider KeySetMetadataProvider { get; set; }
+
+        public void Configure(JwtBearerOptions options)
+        {
+            options.Audience = Configuration["Authentication:Identity:Audience"];
+            options.ConfigurationManager = new WebApplicationConfiguration(null, HttpContextAccessor);
+            var keys = KeySetMetadataProvider.GetKeysAsync().GetAwaiter().GetResult().Keys;
```

Comment:
Yeah, this code shouldn't be in the template, but [we haven't built the integrated web api support](https://github.com/aspnet/IdentityService/issues/93) yet like we've already done for [web apps](https://github.com/aspnet/IdentityService/blob/dev/src/Microsoft.AspNetCore.Identity.Service.IntegratedWebClient/IntegratedWebClientOpenIdConnectOptionsSetup.cs). This class should be moved to an extensions folder for now and then replaced by framework code